components/service/sync-logins/src/test/java/mozilla/components/service/sync/logins/DefaultLoginValidationDelegateTest.kt

```diff
@@ -0,0 +1,203 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.sync.logins
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.TestCoroutineScope
+import kotlinx.coroutines.test.runBlockingTest
+import mozilla.components.concept.storage.Login
+import mozilla.components.concept.storage.LoginValidationDelegate.Result
+import mozilla.components.concept.storage.LoginsStorage
+import mozilla.components.service.sync.logins.DefaultLoginValidationDelegate.Companion.mergeWithLogin
+import mozilla.components.support.test.any
+import mozilla.components.support.test.whenever
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.ArgumentMatchers.anyString
+import org.robolectric.annotation.Config
+
+@RunWith(AndroidJUnit4::class)
+@ExperimentalCoroutinesApi
+class DefaultLoginValidationDelegateTest {
+
+    private lateinit var loginsStorage: LoginsStorage
+    private lateinit var validationDelegate: DefaultLoginValidationDelegate
+    private lateinit var scope: TestCoroutineScope
+
+    @Before
+    @Config(sdk = [21])
+    fun before() {
+        loginsStorage = mockLoginsStorage()
+        scope = TestCoroutineScope()
+        validationDelegate =
+            DefaultLoginValidationDelegate(storage = lazy { loginsStorage }, scope = scope)
+    }
+
+    @Test
+    @Config(sdk = [21])
+    fun `WHEN guid is null or empty THEN should create a new record`() = runBlockingTest {
```

Comment:
What if the record has a `null` guid, but matches an existing record exactly except for a password change?